Remove code migrated to aspnetcore

JunTaoLuo opened the pull request:
TODO after https://github.com/dotnet/aspnetcore/pull/21581 is merged:
- [x] Fixup sln files
- [ ] Invert darc subscription
- [x] Ingest packages from aspnetcore

NTaylorMullen commented on eng/Versions.props:
```diff
@@ -82,8 +82,6 @@
-    <MicrosoftExtensionsHashCodeCombinerSourcesPackageVersion>5.0.0-preview.2.20121.3</MicrosoftExtensionsHashCodeCombinerSourcesPackageVersion>
     <MicrosoftExtensionsNonCapturingTimerSourcesPackageVersion>5.0.0-preview.2.20121.3</MicrosoftExtensionsNonCapturingTimerSourcesPackageVersion>
     <BenchmarkDotNetPackageVersion>0.10.13</BenchmarkDotNetPackageVersion>
     <MicrosoftBuildFrameworkPackageVersion>15.8.166</MicrosoftBuildFrameworkPackageVersion>
```
Later on in this file I believe you can remove all of the `Runtime_` versions now.

JunTaoLuo replied:
Runtime_MicrosoftCodeAnalysisCSharpPackageVersion is still used by Razor.Test.Common which is used by CA.Razor.Workspaces.Test.Common

NTaylorMullen replied:
oh interesting, that's probably a bug and should be replaced with the tooling variant.

JunTaoLuo replied:
I'll try switching it.

JunTaoLuo commented:
I wasn't aware those need to be migrated. I'll handle those in a mop up.

JunTaoLuo commented:
@NTaylorMullen I can move RazorSyntaxGenerator but the benchmarks project depends on Microsoft.CodeAnalysis.Razor.Workspaces. Does it mean that we need to move both of those?

JunTaoLuo commented:
Also, where should the benchmarks move to in aspnetcore? src/Razor/Razor/perf/?

JunTaoLuo commented on src/Razor/src/Microsoft.VisualStudio.Mac.RazorAddin/Microsoft.VisualStudio.Mac.RazorAddin.csproj:
```diff
@@ -42,35 +42,37 @@
-    <MPackFile Include="$(ArtifactsBinDir)Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X\$(Configuration)\netstandard2.0\Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X.pdb" />
-    <MPackFile Include="$(ArtifactsBinDir)Microsoft.AspNetCore.Razor.Language\$(Configuration)\netstandard2.0\Microsoft.AspNetCore.Razor.Language.pdb" />
-    <MPackFile Include="$(ArtifactsBinDir)Microsoft.CodeAnalysis.Razor\$(Configuration)\netstandard2.0\Microsoft.CodeAnalysis.Razor.pdb" />
+    <!-- <MPackFile Include="$(PkgMicrosoft_AspNetCore_Mvc_Razor_Extensions)\lib\netstandard2.0\Microsoft.AspNetCore.Mvc.Razor.Extensions.pdb" /> -->
```
Hmm seems like pdbs are not included in nupkgs.

JunTaoLuo replied:
Looking into options for getting at the pdbs
1. Symbols packages
2. Transport package

I prefer 1 but 2 will definitely work.

JunTaoLuo replied:
@dougbu would it be easy to access symbol packages? If not I can just build a transport package.

dougbu replied:
As you discovered, symbol packages are _only_ uploaded to symbol servers and placed in build-internal artifacts. You did the right thing creating a new package containing what's needed here.

JunTaoLuo commented on src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/Rules/RazorGenerateWithTargetPath.cs:
```diff
@@ -10,8 +10,11 @@
 
 namespace Microsoft.CodeAnalysis.Razor.ProjectSystem.Rules
 {
+    
```
Not sure why these files changed

NTaylorMullen commented:
> @NTaylorMullen I can move RazorSyntaxGenerator but the benchmarks project depends on Microsoft.CodeAnalysis.Razor.Workspaces. Does it mean that we need to move both of those?

Ahh, we may need to split the benchmarks into two. One for aspnetcore and one for aspnetcore-tooling. I don't want to have `Microsoft.CodeAnalysis.Razor.Workspaces` in aspnetcore. Currently the benchmarks only has a [single file](https://github.com/dotnet/aspnetcore-tooling/blob/1b16a37fc76ff0cc2b7a02486fb3eb0289f41738/src/Razor/benchmarks/Microsoft.AspNetCore.Razor.Performance/TagHelperSerializationBenchmark.cs) that depends on the Razor.Workspaces project so everything else can go.

> Also, where should the benchmarks move to in aspnetcore? Based on the naming, it would probably be src/Razor/Razor/perf/ but the rest of the projects under src/Razor/Razor were already in aspnetcore and doesn't have much to do with the benchmarks project, so I think src/Razor/perf might make more sense. Thoughts?

Makes perfect sense to me!

> Hmm seems like pdbs are not included in nupkgs.

We only include the pdbs in the mpacks which might prove to a point of concern in this migration.

JunTaoLuo commented:
I'll need to ingest newer packages and pdbs from aspnetcore when they are available, but otherwise the PR is ready for review.

dougbu commented:
To be clear
> Invert darc subscription

Don't do this until this PR is merged. Also, delete the disabled aspnetcore-tooling -> aspnetcore edge for the 'master' branch now or at the same time. aspnetcore shouldn't need anything from here. If it does, that's a bug to fix before merging this PR.

JunTaoLuo commented on eng/AfterSigning.targets:
```diff
@@ -11,7 +11,7 @@
   <Target Name="_ZipLanguageServerBinaries" AfterTargets="Pack"  Condition=" '$(OS)' == 'Windows_NT' ">
     <!-- This target is defined in eng/targets/Packaging.targets and included in every project. -->
-    <MSBuild Projects="$(RepoRoot)src\Razor\src\Microsoft.AspNetCore.Razor.Language\Microsoft.AspNetCore.Razor.Language.csproj"
+    <MSBuild Projects="$(RepoRoot)src\Razor\src\Microsoft.AspNetCore.Razor.LanguageServer\Microsoft.AspNetCore.Razor.LanguageServer.csproj"
```
This change was made because this target calculates a version number from the M.A.Razor.Language project. However, that project moved, so I changed it to a different project instead.

JunTaoLuo commented on src/Razor/src/Microsoft.VisualStudio.Mac.RazorAddin/Microsoft.VisualStudio.Mac.RazorAddin.csproj:
```diff
@@ -42,35 +42,37 @@
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X" Version="$(MicrosoftAspNetCoreMvcRazorExtensionsVersion1_XPackageVersion)" GeneratePathProperty="true"/>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X" Version="$(MicrosoftAspNetCoreMvcRazorExtensionsVersion2_XPackageVersion)" GeneratePathProperty="true"/>
+    <PackageReference Include="Microsoft.AspNetCore.Razor.Language" Version="$(MicrosoftAspNetCoreRazorLanguagePackageVersion)" GeneratePathProperty="true"/>
+    <PackageReference Include="Microsoft.CodeAnalysis.Razor" Version="$(MicrosoftCodeAnalysisRazorPackageVersion)" GeneratePathProperty="true"/>
```
👍

JunTaoLuo commented on src/Razor/src/Microsoft.VisualStudio.Mac.RazorAddin/Microsoft.VisualStudio.Mac.RazorAddin.csproj:
```diff
@@ -42,35 +42,37 @@
+    <!-- <MPackFile Include="$(PkgMicrosoft_AspNetCore_Mvc_Razor_Extensions_Version1_X)\lib\netstandard2.0\Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X.pdb" /> -->
+    <!-- <MPackFile Include="$(PkgMicrosoft_AspNetCore_Mvc_Razor_Extensions_Version2_X)\lib\netstandard2.0\Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X.pdb" /> -->
+    <!-- <MPackFile Include="$(PkgMicrosoft_AspNetCore_Razor_Language)\lib\netstandard2.0\Microsoft.AspNetCore.Razor.Language.pdb" /> -->
+    <!-- <MPackFile Include="$(PkgMicrosoft_CodeAnalysis_Razor)\lib\netstandard2.0\Microsoft.CodeAnalysis.Razor.pdb" /> -->
```
I'll update this once I have a transport package from aspnetcore to ingest.

JunTaoLuo commented on eng/Versions.props:
```diff
@@ -48,13 +48,19 @@
+    <MicrosoftAspNetCoreMvcRazorExtensionsPackageVersion>5.0.0-preview.6.20276.6</MicrosoftAspNetCoreMvcRazorExtensionsPackageVersion>
+    <MicrosoftAspNetCoreMvcRazorExtensionsVersion1_XPackageVersion>5.0.0-preview.6.20276.6</MicrosoftAspNetCoreMvcRazorExtensionsVersion1_XPackageVersion>
+    <MicrosoftAspNetCoreMvcRazorExtensionsVersion2_XPackageVersion>5.0.0-preview.6.20276.6</MicrosoftAspNetCoreMvcRazorExtensionsVersion2_XPackageVersion>
     <MicrosoftExtensionsConfigurationJsonPackageVersion>5.0.0-preview.6.20271.10</MicrosoftExtensionsConfigurationJsonPackageVersion>
```
Hmm looks like I can only remove Reflection.Metadata.

JunTaoLuo commented on src/Razor/src/Microsoft.VisualStudio.Editor.Razor/LegacyProjectEngineFactory_1_0.cs:
```diff
@@ -16,7 +16,7 @@ internal class LegacyProjectEngineFactory_1_0 : IProjectEngineFactory
         {
             // Rewrite the assembly name into a full name just like this one, but with the name of the MVC design time assembly.
-            var assemblyName = new AssemblyName(typeof(LegacyProjectEngineFactory_1_0).Assembly.FullName);
+            var assemblyName = new AssemblyName(typeof(RazorProjectEngine).Assembly.FullName);
```
@NTaylorMullen made that change so he would know.

NTaylorMullen replied:
> `assemblyName`

Yup, in the next line the name is reset to the correct name. We mostly just do this "FullName" dance to get other information about the assembly

JunTaoLuo commented:
Needs another look @NTaylorMullen @dougbu I've addressed feedback items and updated to consume the latest aspnetcore packages.

ryanbrandenburg commented:
> Needs another look @NTaylorMullen @dougbu I've addressed feedback items and updated to consume the latest aspnetcore packages.
I keep forgetting it exists, but GitHub has a "request re-review" button that's nice for this kind of thing. Sometimes  it still makes sense to make a comment if you need to explain what has changed though.

JunTaoLuo commented:
>Sometimes it still makes sense to make a comment if you need to explain what has changed though.

I did 😄, I've addressed feedback items and updated to consume the latest aspnetcore packages.

JunTaoLuo commented on eng/Versions.props:
```diff
@@ -48,13 +48,19 @@
+    <MicrosoftNETSdkRazorPackageVersion>5.0.0-preview.6.20276.6</MicrosoftNETSdkRazorPackageVersion>
     <SystemDiagnosticsDiagnosticSourcePackageVersion>5.0.0-preview.6.20271.10</SystemDiagnosticsDiagnosticSourcePackageVersion>
     <SystemResourcesExtensionsPackageVersion>5.0.0-preview.6.20271.10</SystemResourcesExtensionsPackageVersion>
     <SystemTextEncodingsWebPackageVersion>5.0.0-preview.6.20271.10</SystemTextEncodingsWebPackageVersion>
```
I cleaned up as much as possible.

NTaylorMullen commented on src/Razor/Razor.NoVsix.sln:
```diff
@@ -10,40 +10,20 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{3C0D6505-79B
 EndProject
```
Technically we don't need the "slim" anymore but we can always clean that up after-the-fact. @JunTaoLuo is already doing us a huge solid by helping to the heavy lifting of this work 😄

JunTaoLuo replied:
Ah, might as well clean that up then, I'm waiting for aspnetcore to ingest dependencies anyway.

NTaylorMullen commented:
Oh actually @JunTaoLuo you should remove the testapps folder:
![image](https://user-images.githubusercontent.com/2008729/83292547-245a6f80-a19f-11ea-9527-f0cf48e98683.png)

And update the `.vscode/settings.json` file to not boot to it

NTaylorMullen commented:
> @NTaylorMullen I was under the impression a lot of the apps are still used elsewhere e.g. https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/test/VSCode.FunctionalTest/tests/TestUtil.ts#L15-L19 so I elected to keep all the tests apps. I can do a more granular check and remove ones that are not referenced I suppose.

Ooh ya look at that. You've learned more about the tooling repo now then I have 😆

dougbu commented on eng/Version.Details.xml:
```diff
@@ -1,62 +1,86 @@
+      <Uri>https://github.com/dotnet/aspnetcore</Uri>
+      <Sha>a799bf68e45c63e62060a7916ebb6d4f87a9aa1b</Sha>
+    </Dependency>
+    <Dependency Name="Microsoft.Extensions.Configuration.Json" Version="5.0.0-preview.6.20277.12" CoherentParentDependency="Microsoft.CodeAnalysis.Razor">
```
This is very weird and is not coherent with the dotnet/aspnetcore 5.0.0-preview.6.20278.12 build. We also don't want to downgrade these dependencies. Suspect you need to run `darc update-dependencies --coherency-only` after we get dotnet/aspnetcore#1947 in.

@mmitche @wtgodbe the dotnet/aspnetcore is using dotnet/runtime dependencies at 5.0.0-preview.6.20271.10 at the a799bf68e45c63e62060a7916ebb6d4f87a9aa1b commit. Why didn't `CoherentParentDependency` do the "correct" thing and back up all that way? We don't really want to downgrade at all but that at least would have been coherent.

mmitche replied:
It did the expected thing when I ran it.

dougbu replied:
You mean it downgraded to 5.0.0-preview.6.20271.10?

dougbu replied:
Oops, I pointed to the wrong dependency update PR.

@JunTaoLuo only just added an arc updating dotnet/aspnetcore directly from dotnet/runtime. I triggered it moments ago. This PR won't be ready to go (really, the dependencies won't be coherent without downgrading) until dotnet/aspnetcore#22370 and the subsequent official build and Maestro++ update are done.

dougbu replied:
@JunTaoLuo you've got lots of approval here but please don't merge until you're able to update the dependencies in this branch correctly.

mmitche replied:
Yep, it downgraded locally.  What's the darc version that got used to do that update? Seems unlikely that would affect things but could be.

JunTaoLuo replied:
Locally for me, darc doesn't seem to do any coherency updates, but it could be because aspnetcore hasn't ingested newer runtime versions yet. I'm waiting to confirm that once the runtime update in aspnetcore is merged.

JunTaoLuo replied:
Triggered a subscription update and looks like the latest commit shows that updates are working as expected.

NTaylorMullen commented:
Woooo!!
